src/librustc/mir/mod.rs

```diff
@@ -1917,6 +1917,15 @@ impl<'tcx> Place<'tcx> {
         }
     }
 
+    /// If this place represents a local variable like `_X` with no
+    /// projections, return `Some(_X)`.
+    pub fn as_local(&self) -> Option<Local> {
```

Comment:
This is only used once, whereas this pattern (usually with the `base` before `projection`) shows up many times in the codebase.

I don't have a strong preference either way but I did want to point it out.

cc @spastorino @oli-obk

Reply:
I don't think it makes sense after Place 2.0 changes, now you can do ...

```rust
if let PlaceBase::Local(l) = place.base {
    something,
}
```

I don't think doing ...

```rust
if let Some(l) = place.as_local() {
    something,
}
```

is any better.

If you prefer that pattern I can remove the function and fix the calls.

Reply:
@spastorino those are not equivalent. I specifically want to ensure that there are no projections.

Reply:
Ohh I was really distracted when reading this, and yes you're right @nikomatsakis I didn't even realize that was something just added.